### ISSUE TITLE
Verified Token Sale Contributions: Ledger Tx Signing Failing

### ISSUE THREAD

**Issue by brianlenz** [CityOfZion/neon-wallet#658]
I'm not entirely sure where this issue lies, but token sale contributions with verification enabled are failing when using Ledger. If you don't use verification, it works perfectly.

With the fixes from @dvdschwrtz / @slipo for #647, verified transactions are otherwise working perfectly, so I believe this is the only outstanding issue remaining with verification.

**Comment by brianlenz** [comment]
@dvdschwrtz / @slipo: I've spent some time looking into this. Here's what I believe to be the issue:

* In neon-wallet, `ledgerNanoSCreateSignatureAsync` properly calculates the signature and serializes the transaction, but it returns a `string`, not an `object`.
* In neon-js, `api.doInvoke` calls `signTx`, which then assigns the `signedTx` returned by the `signingFunction` to `config.tx`
* `attachInvokedContractForMintToken` is then called to attach the invoked contract, but I think it may be resulting in an error due to the fact that it assumes `config.tx` is an `object` not a `string`. Specifically, I believe this will fail:

```
config.tx.scripts.push(attachInvokedContract)
```

I'm not as familiar with the code as you guys are, but I believe the only indication we have that a `mintTokens` function is being invoked is that the `unsignedTx.attributes` array has a usage value of 32 to indicate a script is being invoked. I attempted to come up with a solution, but it doesn't quite seem to work. Commits here:

https://github.com/NarrativeNetwork/neon-wallet/commit/2c9e52893847f82bb7c8f3c50df93c93966bc4c0
https://github.com/brianlenz/neon-js/commit/0b8ba986301ade658c7e99c4e2349b83b098b133

It feels like it's close, but I must be missing something. I can't seem to get any `console.log` statements to work in `attachInvokedContractForMintToken`, so that's not helping with my debugging.

Let me know if you have any thoughts or suggestions!
